service/src/main/java/greencity/service/HabitServiceImpl.java

```diff
@@ -28,56 +32,154 @@
      */
     @Override
-    public HabitVO getById(Long id) {
+    public HabitManagementDto getById(Long id) {
         Habit habit = habitRepo.findById(id)
             .orElseThrow(() -> new WrongIdException(ErrorMessage.HABIT_NOT_FOUND_BY_ID + id));
-        return modelMapper.map(habit, HabitVO.class);
+        return modelMapper.map(habit, HabitManagementDto.class);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public List<HabitDto> getAllHabitsDto() {
-        return habitRepo.findAll()
+    public PageableDto<HabitManagementDto> getAllHabitsDto(Pageable pageable) {
+        Page<Habit> habits = habitRepo.findAll(pageable);
+        List<HabitManagementDto> habitDtos = habitRepo.findAll()
             .stream()
-            .map(habit -> modelMapper.map(habit, HabitDto.class))
+            .map(habit -> modelMapper.map(habit, HabitManagementDto.class))
             .collect(Collectors.toList());
+        return new PageableDto<>(
+            habitDtos,
+            habits.getTotalElements(),
+            habits.getPageable().getPageNumber(),
+            habits.getTotalPages());
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public PageableDto<HabitTranslationDto> getAllHabitsByLanguageCode(Pageable pageable, String language) {
+    public PageableDto<HabitDto> getAllHabitsByLanguageCode(Pageable pageable, String language) {
         Page<HabitTranslation> pages =
             habitTranslationRepo.findAllByLanguageCode(pageable, language);
-        List<HabitTranslationDto> habitTranslationDtos =
+        List<HabitDto> habitTranslationDtos =
             pages.stream()
-                .map(habit -> modelMapper.map(habit, HabitTranslationDto.class))
+                .map(habitTranslation -> modelMapper.map(habitTranslation, HabitDto.class))
                 .collect(Collectors.toList());
         return new PageableDto<>(habitTranslationDtos, pages.getTotalElements(),
             pages.getPageable().getPageNumber(),
             pages.getTotalPages());
     }
-}
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional
+    public HabitManagementDto saveHabitAndTranslations(HabitManagementDto habitManagementDto, MultipartFile image) {
+        Habit habit = habitRepo.save(Habit.builder()
+            .image(habitManagementDto.getImage())
+            .habitTranslations(
+                habitManagementDto.getHabitTranslations().stream()
+                    .map(translationDto -> HabitTranslation.builder()
+                        .description(translationDto.getDescription())
+                        .habitItem(translationDto.getHabitItem())
+                        .name(translationDto.getName())
+                        .language(modelMapper.map(
+                            languageService.findByCode(translationDto.getLanguageCode()),
+                            Language.class)).build())
+                    .collect(Collectors.toList())
+            ).build());
```

Comment:
Looks like Habit builder better to move to some mapper or a least to some private method

Reply:
Is it worth putting this builder in mapper if I'm using mapper + service to set separate field? And my habit is tightly coupled with translations, so I don't know if it makes sense to put it to new private method. Or it is a very bad practise so I have to fix it for sure?

Reply:
Created new card with issue.